mm/2s2h/Rando/MiscBehavior/ClockShuffle.cpp

```diff
@@ -0,0 +1,765 @@
+            u32 timeDiff = (DAY_LENGTH - terminalZeroed) % DAY_LENGTH;
+
+            // Convert to hours (round up to ensure we don't under-calculate)
+            u32 terminalHours = (timeDiff + CLOCK_TIME_HOUR - 1) / CLOCK_TIME_HOUR;
+            totalHours += terminalHours;
+        }
+
+        u32 ownedTime = totalHours * CLOCK_TIME_HOUR;
+
+        // Compensate for the -1 minute offset that vanilla uses
+        // Vanilla uses CLOCK_TIME(6, 0) - 1, Clock Shuffle uses CLOCK_TIME(6, 0)
+        // So we need to add 1 minute to match vanilla behavior
+        ownedTime += CLOCK_TIME_MINUTE;
+
+        // Set the time variable to our calculated value
+        *timeVar = ownedTime;
+    });
+
+    // Hook scarecrow dance time skip to redirect to next owned half-day
+    COND_VB_SHOULD(VB_SCARECROW_DANCE_SET_TIME, IS_RANDO && RANDO_SAVE_OPTIONS[RO_CLOCK_SHUFFLE], {
+        *should = false; // Skip vanilla behavior
+
+        // Calculate next owned half-day after current
+        int currentHalfDay = GetCurrentHalfDayIndex();
+        u8 ownedHalfDaysMask = ClockItems::GetAllOwnedHalfDaysMask();
+        int nextHalfDay = ClockItems::FindNextOwnedHalfDayAfter(currentHalfDay, ownedHalfDaysMask);
+
+        if (nextHalfDay == ClockItems::TERMINAL_STATE) {
+            // Jump to terminal time
+            gSaveContext.save.day = 3;
+            gSaveContext.save.time = GetConfiguredTerminalTime();
+            gSaveContext.respawnFlag = -8; // No daytelop for terminal
+        } else {
+            // Get target half-day configuration
+            const HalfDayTimeConfig* config = GetHalfDayTimeConfig(nextHalfDay);
+            bool isNightHalf = (nextHalfDay % 2 == 1);
+            s32 targetDay = config->dayNumber;
+            u16 targetTime = config->startTime;
+
+            if (isNightHalf) {
+                // Advancing to night - use respawnFlag -8 (no daytelop)
+                gSaveContext.save.day = targetDay;
+                gSaveContext.save.time = targetTime;
+                gSaveContext.respawnFlag = -8;
+            } else {
+                // Advancing to dawn - use respawnFlag -4 (triggers daytelop)
+                // CRITICAL: Subtract 1 from day because daytelop will increment it
+                gSaveContext.save.day = targetDay - 1;
+                gSaveContext.save.time = targetTime;
+                gSaveContext.respawnFlag = -4;
+                SET_EVENTINF(EVENTINF_TRIGGER_DAYTELOP);
+            }
+        }
+    });
+}
+
+// Initialize clock settings and item pool for file creation
+void InitializeFileClocks(std::vector<RandoItemId>& itemPool) {
+    if (!RANDO_SAVE_OPTIONS[RO_CLOCK_SHUFFLE]) {
+        return; // Skip if clocks not enabled
+    }
+
+    int clockMode = RANDO_SAVE_OPTIONS[RO_CLOCK_SHUFFLE_PROGRESSIVE];
+
+    // Check if player has selected any starting time items
+    std::vector<RandoItemId> startingItems = convertStartingItemsToRandoItemId(RANDO_STARTING_ITEMS, ",");
```

Comment:
Need a default, in case somehow the player never touched the starting items setting before. clang format may not like this:
```suggestion
    std::vector<RandoItemId> startingItems = convertStartingItemsToRandoItemId(CVarGetString("gRando.StartingItems", RANDO_STARTING_ITEMS_DEFAULT), ",");
```